lib/experimental/class-gutenberg-rest-global-styles-revisions-controller.php

```diff
@@ -0,0 +1,363 @@
+				$parent->post_author,
+				array(
+					'size' => 24,
+				)
+			);
+		}
+
+		if ( rest_is_field_included( 'author_display_name', $fields ) ) {
+			$data['author_display_name'] = get_the_author_meta( 'display_name', $parent->post_author );
+		}
+
+		if ( rest_is_field_included( 'date', $fields ) ) {
+			$data['date'] = $parent->post_date;
+		}
+
+		if ( rest_is_field_included( 'date_display', $fields ) ) {
+			/* translators: 1: Human-readable time difference, 2: short date combined to show rendered revision date. */
+			$data['date_display'] = sprintf( __( '%1$s (%2$s)', 'gutenberg' ), $time_ago, $date_short );
+		}
+
+		if ( rest_is_field_included( 'date_gmt', $fields ) ) {
+			$data['date_gmt'] = $parent->post_date_gmt;
+		}
+
+		if ( rest_is_field_included( 'id', $fields ) ) {
+			$data['id'] = (int) $item->ID;
+		}
+
+		if ( rest_is_field_included( 'modified', $fields ) ) {
+			$data['modified'] = $parent->post_modified;
+		}
+
+		if ( rest_is_field_included( 'modified_gmt', $fields ) ) {
+			$data['modified_gmt'] = $parent->post_modified_gmt;
+		}
+
+		if ( rest_is_field_included( 'parent', $fields ) ) {
+			$data['parent'] = (int) $parent->ID;
+		}
+
+		if ( rest_is_field_included( 'settings', $fields ) ) {
+			$data['settings'] = ! empty( $config['settings'] ) ? $config['settings'] : new stdClass();
+		}
+
+		if ( rest_is_field_included( 'styles', $fields ) ) {
+			$data['styles'] = ! empty( $config['styles'] ) ? $config['styles'] : new stdClass();
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Retrieves the revision's schema, conforming to JSON Schema.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => "{$this->parent_post_type}-revision",
+			'type'       => 'object',
+			// Base properties for every Revision.
+			'properties' => array(
+
+				/*
+				 * Adds settings and styles from the WP_REST_Revisions_Controller item fields.
+				 * Leaves out GUID as global styles shouldn't be accessible via URL.
+				 */
+				'author'              => array(
+					'description' => __( 'The ID for the author of the revision.', 'gutenberg' ),
+					'type'        => 'integer',
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+				'date'                => array(
+					'description' => __( "The date the revision was published, in the site's timezone.", 'gutenberg' ),
+					'type'        => 'string',
+					'format'      => 'date-time',
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+				'date_gmt'            => array(
```

Comment:
We don't need half of these properties, at least for the purposes of planned UI changes: https://github.com/WordPress/gutenberg/issues/49601

I just added them because they were in the revisions controller.

I've a mind to remove them.

Reply:
Which ones were you thinking of removing?

In principle, I quite like the idea of trying to keep a close parallel between the two revisions endpoints so that there's predictability between the endpoints if someone were to write a plugin or other tool that wants to call the endpoints — but I have only just started looking at this PR 😄

Reply:
Good question. I was wondering whether to include only a minimal set of properties that will serve to create the revisions UI (see https://github.com/WordPress/gutenberg/issues/49601), and leave it open for extension later, but you're right about predictability. There's no harm in leaving them in. 👍

Reply:
> Which ones were you thinking of removing?

I've already left out slug, but was thinking of removing dates (except for the display one).

Do you reckon I should reinstate slug? I guess it could be at least used as a key for iterable components

Here's a raw preview (ignore the rendered content)

<img width="874" alt="Screenshot 2023-04-24 at 12 08 45 pm" src="https://user-images.githubusercontent.com/6458278/233884105-a7522723-1ea5-4c32-8900-17f1a1544771.png">

Reply:
> I've already left out slug, but was thinking of removing dates (except for the display one).

Personally, I'd keep all the dates ones so that they're filterable / sortable by the consumer.

> I've already left out slug, but was thinking of removing dates (except for the display one).

We've already got `id`, so slug might not be all that useful, especially since `title` isn't present in the response, and users aren't able to update the title of the global styles entry, anyway.

I quite like the balance that you've already got in this PR, it's looking quite good to me 🙂